c/src/atom.rs

```diff
@@ -271,36 +411,36 @@ pub extern "C" fn bindings_set_eq(set: *const bindings_set_t, other: *const bind
 /// string terminator.
 #[no_mangle]
 pub extern "C" fn bindings_set_to_str(set: *const bindings_set_t, buf: *mut c_char, buf_len: usize) -> usize {
-    let set = unsafe{ &(*set).set };
+    let set = unsafe{ (&*set).borrow() };
     write_into_buf(set, buf, buf_len)
 }
 
 #[no_mangle]
 pub extern "C" fn bindings_set_is_empty(set: *const bindings_set_t) -> bool {
-    let set = unsafe{ &(*set).set };
+    let set = unsafe{ (&*set).borrow() };
     set.is_empty()
 }
 
 #[no_mangle]
 pub extern "C" fn bindings_set_is_single(set: *const bindings_set_t) -> bool {
-    let set = unsafe{ &(*set).set };
+    let set = unsafe{ (&*set).borrow() };
     set.is_single()
 }
 
 #[no_mangle]
 pub extern "C" fn bindings_set_iterate(set: *mut bindings_set_t, callback: bindings_mut_callback_t, context: *mut c_void) {
-    let set = unsafe{ &mut (*set).set };
+    let set = unsafe{ (&mut *set).borrow_mut() };
```

Comment:
I don't understand what you mean?  Do you mean you don't like the style of the `borrow ` / `borrow_mut` functions?

I chose this as the "preamble" of the functions because the first thing they do is convert the c type into the Rust type that the rest of the code can work with.